istioctl/cmd/tag.go

```diff
@@ -0,0 +1,400 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+	admit_v1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"istio.io/api/label"
+)
+
+const (
+	// TODO(monkeyanator) move into istio/api
+	istioTagLabel             = "istio.io/tag"
+	istioInjectionWebhookName = "sidecar-injector.istio.io"
+	revisionTagNamePrefix     = "istio-revision-tag"
+)
+
+var (
+	// revision to point tag webhook at
+	revision         = ""
+	overwrite        = false
+	skipConfirmation = false
+)
+
+func tagCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tag",
+		Short: "Command group used to interact with revision tags",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.HelpFunc()(cmd, args)
+			if len(args) != 0 {
+				return fmt.Errorf("unknown subcommand %q", args[0])
+			}
+
+			return nil
+		},
+	}
+
+	cmd.AddCommand(tagSetCommand())
+	cmd.AddCommand(tagListCommand())
+	cmd.AddCommand(tagRemoveCommand())
+
+	return cmd
+}
+
+func tagSetCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:        "set",
+		Short:      "Create or modify revision tags",
+		Example:    "istioctl x tag set prod --revision 1-8-0",
```

Comment:
The Example should be big, with multiple examples and comments.  Use something like _kubeinject.go_ as a model.  There should also be a Long.

If you do `docker tag --help` it says something like "Usage: docker tag SOURCE_IMAGE[:TAG] TARGET_IMAGE[:TAG] / Create a tag TARGET_IMAGE that refers to SOURCE_IMAGE".  The Long needs something like to make it clear what the tags refer to control planes.  "Tag an Istio control plane revision for use with namespace istio.io/rev= injection labels".